client.go

```diff
@@ -162,12 +172,32 @@ var h2Transport = http2.Transport{
 // NewClient creates a RESTful client instance.
 // The instance has a semi-permanent transport TCP connection.
 func NewClient() *Client {
+	return NewClientWInterface("")
+}
+
+// NewClientWInterface creates a RESTful client instance with binded to that interface.
+// The instance has a semi-permanent transport TCP connection.
+func NewClientWInterface(theInterface string) *Client {
 	t := http.DefaultTransport.(*http.Transport).Clone()
 	t.MaxIdleConns = 100
 	t.MaxConnsPerHost = 100
 	t.MaxIdleConnsPerHost = 100
 	dialer := &net.Dialer{Timeout: 2 * time.Second, KeepAlive: 30 * time.Second}
-	t.DialContext = dialer.DialContext
+	if theInterface != "" {
+		IPs := getIpFromInterface(theInterface)
+		t.DialContext = func(ctx context.Context, network, addr string) (net.Conn, error) {
+			dialer.LocalAddr = IPs.IPv4
+			conn, err := dialer.DialContext(ctx, network, addr)
+			// if there is IPv6 address and err and error is no suitable address found than try it with IPv6
+			if IPs.IPv6 != nil && err != nil && reflect.DeepEqual(err, &net.OpError{Op: "dial", Net: network, Source: nil, Addr: nil, Err: &net.AddrError{Err: "no suitable address found", Addr: IPs.IPv4.String()}}) {
```

Comment:
This error checking looks a bit fragile. Is this deep checking really needed?

Reply:
It would be nice to have a better solution. I won't like to always check if the IPv6 is working or not, because can hide the real connection error. Also errors.Is function doesn't work on FT.

Reply:
Yeah, but this `DeepEqual` can break at any moment. Even checking for substring in `about err.Error()` sounds better.